Study/task1.rb

```diff
@@ -0,0 +1,120 @@
+class Station
+ attr_reader :trainlist
+  def initialize(name)
+    @name = name
+    @trainlist = []
+  end
+  
+  def take_train(train)
+    @trainlist.push(train.number)
+    @trainlist.push(train.type)
+    @trainlist.push(train.number_of_vagons)
+  end
+  
+  def delete_train(train)
+    @trainlist.delete(train.number)
+    @trainlist.delete(train.type)
+    @trainlist.delete(train.number_of_vagons)
+  end
+  
+  def count(type)
+    count_pass = 0
+    count_gruz = 0
+    i = @trainlist.count{|x| x == "pass"}
+    k = @trainlist.count{|x| x == "gruz"} 
+   if type == "pass"
+    return i 
+    else return k
+   end
+  end
+ end
+
+class Route
+  attr_reader :stations
+  def initialize(first_station, last_station)
+    @stations = [first_station, last_station]
+  end
+  
+  def add_st(n=1,station)
+     @stations.insert(n, station)
+  end
+  
+  def delete_st(station)
+     @stations.delete(station)
+  end
+  
+  def show_stations_names
+    puts @stations
+  end 
+end
+
+class Train
+  
+  attr_reader :number_of_vagons, :type, :number
+  
+  def initialize(number, type, number_of_vagons)
+    @number = number
+    @type = type
+    @number_of_vagons = number_of_vagons
+  end
+  
+  def speed_up(speed)
+    @speed -= speed
+  end
+  
+  def decrease_speed(value)
+    if @speed > value
+     @speed -= value
+    end
+  end
+  
+  def drop_vagon
+   if @speed == 0
+    if @number_of_vagons > 0
+     @number_of_vagons -=1
+     end
+    end
+  end
+  
+  def take_vagon
+    if @speed == 0
+     @number_of_vagons += 1
+    end
+  end
+  
+  def route_add(route)
+    @routename = route

```

Comment:
```suggestion
    @route = route
```
переменная хранить объект маршрута целиком, а не его имя, нужно именно так и назвать ее, чтобы не путать себя и других